dealership/contracts.py

```diff
@@ -4,9 +4,29 @@ class Contract(object):
 
 class BuyContract(Contract):
     def __init__(self, vehicle, customer, monthly_payments):
-        pass
+        self.vehicle = vehicle
+        self.customer = customer
+        self.monthly_payments = monthly_payments
+    
+    def total_value(self):
+        interest = self.vehicle.interest_rate * self.monthly_payments * self.vehicle.sale_price() / 100
+        discount_multiplier = 1 - (self.customer.discount if self.customer.is_employee() else 0)
+        return discount_multiplier * (self.vehicle.sale_price() + interest)
```

Comment:
You're computing `self.vehicle.sale_price()`. If this is an expensive operation it might hurt. It's not in this case, so it's totally fine.